src/Microsoft.DocAsCode.Common/Git/GitUtility.cs

```diff
@@ -59,7 +59,31 @@ public static GitRepoInfo GetRepoInfo(string directory)
                 throw new GitException("Can't find git command in current environment");
             }
 
-            return Cache.GetOrAdd(directory, GetRepoInfoCore);
+            if (IsGitRoot(directory))
+            {
+                return Cache.GetOrAdd(directory, GetRepoInfoCore);
+            }
+
+            var parentDirInfo = Directory.GetParent(directory);
+            return Cache.GetOrAdd(directory, GetRepoInfo(parentDirInfo?.FullName));
```

Comment:
Yes, `Directory.GetParent` will throw the exception.

---
In reply to: [94093186](https://github.com/dotnet/docfx/pull/1077#discussion_r94093186) [](ancestors = 94093186)